whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotFileGenerator.java

```diff
@@ -45,51 +48,56 @@ public SnapshotFileGenerator(
+//                return Optional.empty();
+//            } else {
+//                LOGGER.info("Not generating snapshot file since no deltas have been published since v{} with serialID {}",
+//                    version.getVersion(), version.getLastSerialId());
+//                return Optional.empty();
+//            }
         }
-        LOGGER.info("now at version: {}", version);
-        if (version.getVersion() > 1) {
-            LOGGER.debug("not syncing deltas to snapshot");
-            return Optional.empty();
-        }
-        final PublishableSnapshotFile snapshotFile = new PublishableSnapshotFile(version);
-        final String fileName = NrtmFileUtil.newFileName(snapshotFile);
-        final Stopwatch stopwatch = Stopwatch.createStarted();
-        try (final OutputStream out = nrtmFileStore.getFileOutputStream(snapshotFile.getSessionID(), fileName)) {
-            snapshotFileSerializer.writeSnapshotAsJson(snapshotFile, out);
-        }
-        try {
-            final String sha256hex = DigestUtils.sha256Hex(nrtmFileStore.getFileInputStream(snapshotFile.getSessionID(), fileName));
-            snapshotFileRepository.save(
-                snapshotFile.getVersionId(),
-                fileName,
-                sha256hex
-            );
-            snapshotFile.setFileName(fileName);
-            snapshotFile.setHash(sha256hex);
-            LOGGER.info("Generated snapshot file in {}", stopwatch);
-            return Optional.of(snapshotFile);
-        } catch (final IOException e) {
-            LOGGER.error("Exception thrown when calculating hash of snapshot file", e);
-            throw e;
+        final List<PublishableSnapshotFile> snapshotFiles = new ArrayList<>();
+        for (final NrtmVersionInfo version : versions) {
+            LOGGER.info("{} at version: {}", version.getSource().source(), version);
+            if (version.getVersion() > 1) {
+                LOGGER.debug("Sync Whois changes to snapshot here (not implemented)");
+            }
+            final PublishableSnapshotFile snapshotFile = new PublishableSnapshotFile(version);
+            final String fileName = NrtmFileUtil.newFileName(snapshotFile);
+            Stopwatch stopwatch = Stopwatch.createStarted();
+            try (final OutputStream out = nrtmFileStore.getFileOutputStream(snapshotFile.getSessionID(), fileName)) {
+                snapshotFileSerializer.writeSnapshotAsJson(snapshotFile, out);
+            }
+            LOGGER.info("Wrote JSON for {} in {}", snapshotFile.getSource().name(), stopwatch);
+            try {
+                stopwatch = Stopwatch.createStarted();
+                final String sha256hex = DigestUtils.sha256Hex(nrtmFileStore.getFileInputStream(snapshotFile.getSessionID(), fileName));
+                snapshotFile.setFileName(fileName);
+                snapshotFile.setHash(sha256hex);
+                snapshotFileRepository.insert(snapshotFile);
```

Comment:
Here you are inserting one file per source in the end. However, I see that you are not using the source to identify that file, are we going to use the hash to identify the file of that source, right? there is going to be a match between the source and the hash